akka-actor/src/main/scala/akka/io/dns/DnsSettings.scala

```diff
@@ -44,6 +46,50 @@ private[dns] final class DnsSettings(system: ExtendedActorSystem, c: Config) {
 
   val ResolveTimeout: FiniteDuration = c.getDuration("resolve-timeout").asScala
 
+  private lazy val resolvConf: Option[ResolvConf] = {
+    val etcResolvConf = new File("/etc/resolv.conf")
+    // Avoid doing the check on Windows, no point
+    if (sys.props.get("os.name").exists(os ⇒ os.startsWith("Windows") || os.startsWith("WINDOWS"))) {
+      None
+    } else if (etcResolvConf.exists()) {
+      val parsed = ResolvConfParser.parseFile(etcResolvConf)
+      parsed match {
+        case Success(value) ⇒ Some(value)
+        case Failure(exception) ⇒
+          if (system.log.isWarningEnabled) {
+            system.log.error(exception, "Error parsing /etc/resolv.conf, ignoringto")
```

Comment:
check `isWarningEnabled` but `log.error` is used. Is that on purpose?

Reply:
Also, the message seems incomplete: `"..., ignoringto"`

Reply:
It's on purpose - the only way to log an exception in Akka (and get that exception to be passed as an exception to the underlying log framework and logged according to that log frameworks configuration, eg with the number of stack trace lines configured by that framework) is to log it at `error`, since Akka's logger doesn't have `warn`, `info` or `debug` methods that accepts exceptions. So the next best thing is to check that the log level that you actually want it logged at is enabled before logging it at error - this is done throughout Akka.

And the message is a mistake, I'll fix.

Reply:
> So the next best thing is to check that the log level that you actually want it logged at is enabled before logging it at error - this is done throughout Akka.

How confusing.  I wonder why this is the case.

Reply:
Not adding methods to a public interface. Typed has stack traces for levels other than error